processing/src/main/java/org/apache/druid/query/filter/InDimFilter.java

```diff
@@ -302,10 +304,14 @@ public BitmapColumnIndex getBitmapColumnIndex(ColumnIndexSelector selector)
 
       if (indexSupplier == null) {
         // column doesn't exist, match against null
-        return Filters.makeMissingColumnNullIndex(
-            predicateFactory.makeStringPredicate().apply(null),
-            selector
-        );
+        DruidPredicateMatch match = predicateFactory.makeStringPredicate().apply(null);
```

Comment:
Move this logic into `Filters.makeMissingColumnNullIndex`, and change its signature to `makeMissingColumnNullIndex(DruidPredicateMatch matchesNull, final ColumnIndexSelector selector)`?

Then this code would be a bit simpler, it would remain a call to `Filters.makeMissingColumnNullIndex`.

Reply:
heh i did that at first but were only the 3 callers so seemed barely worth having, then again after sleeping on it i agree, it is just enough callers so will add it back